apollo-router-core/src/traits.rs

```diff
@@ -76,14 +122,16 @@ impl<T: ?Sized> WithCaching for T where T: QueryPlanner + Sized + 'static {}
 /// a [`Response`] that can be returned immediately to the user. This is because GraphQL does
 /// not use the HTTP error codes, therefore it always return a response even if it fails.
 #[async_trait::async_trait]
-pub trait Router<T: PreparedQuery>: Send + Sync + Debug {
-    async fn prepare_query(&self, request: &Request) -> Result<T, Response>;
+pub trait Router: Send + Sync + Debug {
+    type PreparedQuery: PreparedQuery;
+
+    async fn prepare_query(&self, request: &Request) -> Result<Self::PreparedQuery, Response>;
 }
 
 /// An object that can be executed to return a [`Response`].
 #[async_trait::async_trait]
 pub trait PreparedQuery: Send + Debug {
-    async fn execute(self, request: Arc<Request>) -> Response;
+    async fn execute(self, request: RouterRequest) -> Response;
```

Comment:
The `Request` type of apollo-router-core is already scoped under `graphql::`, so you use `graphql::Request` to get a "graphql request". Adding the prefix Router here is redundant and contradicting with the decision we took to use the `graphql::` scope. (That was before you arrive @Geal )

Reply:
Never mind this comment. I got it